crypto/keys/internal/ecdsa/privkey_internal_test.go

```diff
@@ -63,4 +64,29 @@ func (suite *SKSuite) TestSign() {
 	// Mutate the message
 	msg[1] ^= byte(2)
 	require.False(suite.pk.VerifySignature(msg, sig))
+
+	// mutate the signature by scalar neg'ing the s value
+	// to give a high-s signature, valid ECDSA but should
+	// be invalid with Cosmos signatures.
+	// code mostly copied from privkey/pubkey.go
+
+	// extract the r, s values from sig
+	r := new(big.Int).SetBytes(sig[:32])
+	low_s := new(big.Int).SetBytes(sig[32:64])
+
+	// flip the s value into high order of curve P256
+	// leave r untouched!
+	high_s := new(big.Int).Mod(new(big.Int).Neg(low_s), elliptic.P256().Params().N)
+
+	// turn these big ints back into 0-padded bytes
+	// replicating what happens in signature
+	rBytes := r.Bytes()
+	sBytes := high_s.Bytes()
+	sigBytes := make([]byte, 64)
+
+	// 0 pad the byte arrays from the left if they aren't big enough.
+	copy(sigBytes[32-len(rBytes):32], rBytes)
+	copy(sigBytes[64-len(sBytes):64], sBytes)
+
+	require.False(suite.pk.VerifySignature(msg, sigBytes))
```

Comment:
```suggestion
	require.True(ecdsa.Verify(pk, msg, r, high_s)) // not sure of syntax
	require.False(suite.pk.VerifySignature(msg, sigBytes))
```

Reply:
Clever, but I can't figure out how to convert the types for the key:

        // check whether msg can be verified with same key, and high_s
	// value using "regular ecdsa signature
	require.True(ecdsa.Verify(ecdsa.PublicKey(suite.pk), msg, r, high_s))

Doesn't work - I can't seem to convert from a PubKey to an ecdsa.PublicKey even though one struct contains the other.

Unless I use unsafe pointers. 
	// check whether msg can be verified with same key, and high_s
	// value using "regular ecdsa signature
	ecdsaKey := *(*ecdsa.PublicKey)(unsafe.Pointer(&suite.pk))
	require.True(ecdsa.Verify(&ecdsaKey, msg, r, high_s))

And then, I'm pretty sure I get the "normal" verify for that key, which is doing the low_s normalization check anyway...

Reply:
I think this might work
```
ecdsa.Verify(&suite.pk.PublicKey, crypto.Sha256(msg), r, high_s)
```

Reply:
It does not:
`./privkey_internal_test.go:95:28: cannot use &suite.CommonSuite.pk (type *PubKey) as type *ecdsa.PublicKey in argument to ecdsa.Verify`

I've tried several different ways of casting, including the only way that the cast works (direct unsafe pointer). Only the pointer cast works, but that gives me the in-memory representation of the exact same key, which means it's calling the normalized-s verify anyway, so not making a proper test. I suppose it might be possible to take the modified public key, marshal it to bytes, and then create the ecdsa.PublicKey from bytes and call verify on that? Will check that...

Reply:
Figure the test out and it's working as expected. Resolving this.